### PR TITLE
115 selective disclosure ssi update

### PR DESCRIPTION
**Pull request by edchapman88**

### CODE REVIEW

File: trustchain-api/src/api.rs [hunk 1]
```diff
@@ -329,6 +351,113 @@ mod tests {
     }
 
     #[ignore = "requires a running Sidetree node listening on http://localhost:3000"]
+    #[tokio::test]
+    async fn test_verify_rss_credential() {
+        // **NOT** using init(), because the chain written to a temp dir in init() does not have any
+        // RSS keys.
+        // init();
```
Comment (edchapman88) [inline trustchain-api/src/api.rs:358]: Set the keys up such that init can be called as with the other tests.

File: trustchain-api/src/api.rs [hunk 2]
```diff
@@ -444,4 +573,16 @@ mod tests {
...
+    fn get_key_entry() {
+        use ps_sig::keys::Params;
+        use ssi::jwk::rss::generate_keys_jwk;
+        use ssi::jwk::JWK;
+
+        let key: JWK = generate_keys_jwk(64, &Params::new("test".to_string().as_bytes())).unwrap();
+        println!("{}", serde_json::to_string_pretty(&key).unwrap());
+        let entry: PublicKeyEntry = key.try_into().unwrap();
+        println!("{}", serde_json::to_string_pretty(&entry).unwrap());
+    }
```
Comment (edchapman88) [inline trustchain-api/src/api.rs:587]: Refer to this test in the Dev Guide somewhere? It's used to generate the key entry format put onto IPFS for an expandable public key.

File: trustchain-ffi/src/mobile.rs [hunk 3]
```diff
@@ -153,6 +155,32 @@ pub fn vc_verify_credential(credential: String, opts: String) -> Result<String>
...
+    let resolver =
+        trustchain_resolver_light_client(&endpoint_opts.trustchain_endpoint().to_address());
+
+    let rt = Runtime::new().unwrap();
+    rt.block_on(async {
+        o_cred
+            .rss_redact(masked_copy, &resolver, &mut ContextLoader::default())
+            .await
+    })
+    .map_err(|err| FFIMobileError::FailedToRedactCredential(err))?;
```
Comment (edchapman88) [inline trustchain-ffi/src/mobile.rs:180]:
```suggestion
    .map_err(FFIMobileError::FailedToRedactCredential)?;
```

File: trustchain-http/src/issuer.rs [hunk 4]
```diff
@@ -108,11 +110,16 @@ impl TrustchainIssuerHTTP for TrustchainIssuerHTTPHandler {
             }
         }
         let issuer = IONAttestor::new(issuer_did);
+        let key_id = if rss {
+            Some("Un2E28ffH75_lvA59p7R0wUaGaACzbg8i2H9ksviS34")
+        } else {
+            None
+        };
```
Comment (edchapman88) [inline trustchain-http/src/issuer.rs:117]: Don't refer to a specific thumbprint. (I thought we had already fixed this?)

File: trustchain-api/Cargo.toml [hunk 5]
```diff
@@ -8,12 +8,17 @@ edition = "2021"
 [dependencies]
 trustchain-core = { path = "../trustchain-core" }
 trustchain-ion = { path = "../trustchain-ion" }
+ps_sig = { git = "https://github.com/alan-turing-institute/RSS.git", branch = "new_rss" }
 
 async-trait = "0.1"
 serde_json = "1.0"
-ssi = {git="https://github.com/alan-turing-institute/ssi.git", branch="modify-encode-sign-jwt", features = ["http-did", "secp256k1"]}
-did-ion = {git="https://github.com/alan-turing-institute/ssi.git", branch="modify-encode-sign-jwt"}
+ssi = { git = "https://github.com/alan-turing-institute/ssi.git", branch = "1-rss", features = [
```
Comment (edchapman88) [inline trustchain-api/Cargo.toml:15]:
```suggestion
ssi = { git = "https://github.com/alan-turing-institute/ssi.git", rev = "1aa3223a384ee71df1333bbce04af445e852eab5", features = [
```

**Review by thobson88 — APPROVED** [review]
LGTM

**Pull request merged**